localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1962,6 +1998,8 @@ def validate_event_source_mapping(self, context, request):
         fn = state.functions.get(function_name)
         if not fn:
             raise InvalidParameterValueException("Function does not exist", Type="User")
+
+        self.check_service_resource_exists(service, request["EventSourceArn"])
```

Comment:
the `EventSourceArn` parameter of the incoming request is not a required parameter: it is not given when the event source is an external Apache Kafka cluster. This would raise a `KeyError`, so it is probably safer to do a check before.
```suggestion
        if source_arn := request.get("EventSourceArn"):
            self.check_service_resource_exists(service, source_arn)
```